cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/TransparentProxy.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    /**
+     * Verifies if the destination is found by making a HEAD HTTP request.
+     *
+     * @param destination
+     *            the destination to use
+     * @param destinationName
+     *            the name of the destination to check
+     * @return true if the destination is not found, false otherwise
+     */
+    private static
+        boolean
+        isDestinationNotFound( @Nonnull final TransparentProxyDestination destination, final String destinationName )
+    {
+        final HttpClient httpClient = HttpClientAccessor.getHttpClient(destination);
+        final URI destinationUri = destination.getUri();
+        final HttpHead headRequest = new HttpHead(destinationUri);
+        log.debug("Making HEAD request to check if destination {} is found...", destinationName);
+        final HttpResponse response;
+        try {
+            response = httpClient.execute(headRequest);
+        }
+        catch( IOException e ) {
+            log.debug("HEAD request to destination {} failed with exception: {}", destinationName, e.getMessage(), e);
+            return false;
+        }
+        final int statusCode = response.getStatusLine().getStatusCode();
+
+        boolean destinationNotFound = false;
+        if( statusCode == HttpStatus.SC_BAD_GATEWAY ) {
+            final String errorInternalCode = getHeaderValue(response, X_ERROR_INTERNAL_CODE_HEADER);
+            if( Integer.toString(HttpStatus.SC_NOT_FOUND).equals(errorInternalCode) ) {
+                destinationNotFound = true;
+            }
+        }
+
+        log
+            .debug(
+                "HEAD request to destination {} returned status code: {}, x-error-internal-code: {}, found: {}",
+                destinationName,
+                statusCode,
+                getHeaderValue(response, X_ERROR_INTERNAL_CODE_HEADER),
+                !destinationNotFound);
+
+        return destinationNotFound;
+    }
+
+    /**
+     * Helper method to extract header value from HTTP message.
+     *
+     * @param message
+     *            the HTTP message
+     * @param headerName
+     *            the name of the header to extract
+     * @return the header value if present, "" otherwise
+     */
+    private static String getHeaderValue( @Nonnull final HttpMessage message, @Nonnull final String headerName )
+    {
+        if( message.containsHeader(headerName) ) {
+            return message.getFirstHeader(headerName).getValue();
+        }
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public Try<Destination> tryGetDestination( @Nonnull final String destinationName )
+    {
+        final TransparentProxyDestination destination =
+            TransparentProxyDestination.gateway(destinationName, uri).build();
+
+        if( isDestinationNotFound(destination, destinationName) ) {
+            return Try.failure(new DestinationAccessException("Destination not found: " + destinationName));
```

Comment:
It's critical that `DestinationNotFoundException` is thrown, otherwise the chain will not try further loaders.

```suggestion
            return Try.failure(new DestinationNotFoundException("Destination '" + destinationName + "' could not be resolved for the current tenant " + tenant + " by Transparent Proxy running on " + uri));
```

Also, we should include more information in the exception, otherwise users have little information for troubleshooting

Reply:
Changed to DestinationNotFoundException. Message would be retrieved from TP "X-Error-Message" header:
"Destination 'google44' not found in tenant 'tp-integration-tests-8imc9i7e'. Contact the Destination Administrator."
or if fragment exists
Destination 'google' with fragment 'blalala' not found in tenant 'tp-integration-tests-8imc9i7e'. Contact the Destination Administrator.

if these are okay